### ISSUE TITLE
LinkEntryAsync doesn't work as expected

### ISSUE THREAD

**Issue by nycoss13** [object/Simple.OData.Client#277]
Hi,

I have a one-to-many relation between Entity and Techlog

I have some troubles using LinkEntryAsync:
```

            var entities = await _oDataClient
                .For<Entity>(SharedVariables.Fullname(typeof(Entity)))
                .FindEntriesAsync();
            var entity = entities.FirstOrDefault();
            var guid = entity.Oid;

            var addedTechlog = await _oDataClient
            .For<Techlog>(SharedVariables.UpdateFullname(typeof(Techlog)))
            .Set(new
            {
                Oid = Guid.NewGuid(),
                Reference = DateTime.Now.ToString(),
                Date = DateTime.Now,
                Type = TechlogType.Electronic,
                Status = TechlogStatus.Valid,
                //Entity = ....to be set later
            })
            .InsertEntryAsync(true);

            await _oDataClient
                .For<Techlog>(SharedVariables.UpdateFullname(typeof(Techlog)))
                .Key(addedTechlog.Oid)
                .LinkEntryAsync(x => x.Entity, entity);

```
which sends the following to the server (and fails):

```
PUT http://blueeye-mobile.cloudapp.net/BlueEye.DataService.svc/BlueEye_BusinessObjectsLibrary_Techlog(guid%27bdf63023-5c0a-4ce1-ad48-6e2138f6e1ab%27)/$links/Entity HTTP/1.1
Prefer: return-no-content
userCookie: .....
Content-Type: application/json; charset=utf-8
Host: blueeye-mobile.cloudapp.net
Content-Length: 114
Expect: 100-continue

{
  "url":"http://blueeye-mobile.cloudapp.net/BlueEye.DataService.svc/BlueEye_BusinessObjectsLibrary_Entity()"
}
```

I know I am supposed to send exactly the following request to make it work:

```
PUT /BlueEye.DataService.svc/BlueEye_BusinessObjectsLibrary_BusinessObjectBase/BlueEye_BusinessObjectsLibrary_Techlog(guid'BD8B1FAA-61D5-409A-9DCB-F2C3165545B3')/$links/Entity HTTP/1.1
HOST: blueeye-mobile.cloudapp.net
usercookie: .....
content-type: application/json
content-length: 235
 
{
	"url": "http://blueeye-mobile.cloudapp.net/BlueEye.DataService.svc/BlueEye_BusinessObjectsLibrary_BusinessObjectBase/BlueEye_BusinessObjectsLibrary_Entity(guid'26546ec4-8f97-4aad-9351-b2a06202f2a6')"
 
}
```

How am I supposed to use LinkEntryAsync to make the request content as:
http://blueeye-mobile.cloudapp.net/BlueEye.DataService.svc/BlueEye_BusinessObjectsLibrary_BusinessObjectBase/BlueEye_BusinessObjectsLibrary_Entity(guid'26546ec4-8f97-4aad-9351-b2a06202f2a6')
respecting inherited classes namespace and be able to pass the guid to the Entity collection?

thanks in advance

**Comment by object** [comment]
Hi,
I suspect this might have to do with your resource hierarchy - both entity types are subclasses of some base type. I'd like to test this further. I see you have an endpoint that I can reach. Can you allow me to test it, so I can debug my code? If so, the easiest for me would be if you can provide me with class definitions of the types you are using in your code (Techlog, Entity etc.)

**Comment by nycoss13** [comment]
Hi,

Yes I can do this. Is it possible to make that tcket private or can you provide me with your email?

**Comment by object** [comment]
You can send details to my email: vagif.abilov at gmail.

**Comment by object** [comment]
I believe it should work now, released version 4.23. Please check it out.

**Comment by nycoss13** [comment]
Awesome! Many thanks Vagif!
